app/models/work.rb

```diff
@@ -454,9 +454,13 @@ def challenge_claim_ids
 
   # Only allow a work to fulfill an assignment assigned to one of this work's authors
   def challenge_assignment_ids=(ids)
+    valid_ids = ids.reject(&:blank?)
```

Comment:
Since you're using `where` below, instead of calling `find`, you don't need to filter the IDs to remove the empty string here. You should just be able to use `where(id: ids)`.

(If you want to preserve the existing behavior of a 404 when passing an invalid assignment ID, you could keep this line and replace `ChallengeAssignment.where(id: valid_ids)` below with `ChallengeAssignment.find(valid_ids)`. But given that this function is already built around ignoring invalid input, `where` is probably the better approach.)